### PR TITLE
CC-29651 Click&Collect integration.

### PR DESCRIPTION
**Pull request by romanhavrylko**
#### Overview

- Ticket: https://spryker.atlassian.net/browse/CC-29651

### CODE REVIEW

**Review by abitskil — COMMENTED** [review]
Looks good as the early integration, requires an additional round of review when the feature is stabilized. 

Notes:
* Data import files have to be substituted by demo shop specific ones.

File: src/Pyz/Yves/ShopUi/Theme/default/components/molecules/main-popup/main-popup.scss [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        height: rem(24);
+        right: rem(20);
+        top: rem(20);
+        transition: none;
+        color: $setting-color-black;
+
+        .icon {
+            width: 100%;
+            height: 100%;
+        }
```
Comment (FRITS-Kh) [inline src/Pyz/Yves/ShopUi/Theme/default/components/molecules/main-popup/main-popup.scss:30]: Please use a modifier in the `icon` instead.

File: src/Pyz/Yves/ShopUi/Theme/default/components/molecules/search-form/search-form.twig [hunk 2]
```diff
@@ -10,14 +10,26 @@
...
             <div class="col col--sm-10">
-                <input name="q" type="text" class="suggest-search__input suggest-search__input--expand {{ config.jsName }}__input--{{ attributes['data-search-id'] }}" placeholder="{{ 'global.search' | trans }}" value="{{data.searchValue}}" autofocus="autofocus" autocomplete="off">
+                <input 
+                    name="q" 
+                    type="text" 
+                    class="suggest-search__input suggest-search__input--expand {{ config.jsName }}__input--{{ attributes['data-search-id'] }}" 
+                    placeholder="{{ 'global.search' | trans }}" 
+                    value="{{ data.searchValue }}" 
+                    autofocus="autofocus" 
+                    autocomplete="off"
```
Comment (FRITS-Kh) [inline src/Pyz/Yves/ShopUi/Theme/default/components/molecules/search-form/search-form.twig:31]:
```suggestion
                    name="q"
                    class="suggest-search__input suggest-search__input--expand {{ config.jsName }}__input--{{ attributes['data-search-id'] }}"
                    placeholder="{{ 'global.search' | trans }}"
                    value="{{ data.searchValue }}"
                    autocomplete="off"
                    autofocus
```

**Review by FRITS-Kh — CHANGES_REQUESTED** [review]

File: src/Pyz/Yves/CheckoutPage/Theme/default/components/molecules/address-item-form-field-list/address-item-form-field-list.twig [hunk 3]
```diff
@@ -56,36 +69,35 @@
...
+                                        servicePointClassName: embed.itemShippingClassName,
                                     },
-                                }) }}
+                                } only %}{% endwidget %}
+
+                                <div class="{{ embed.deliveryContainerClassName }} {{ embed.shipmentTypeHiddenClassName }}">
+                                    {{ form_row(embed.item.shippingAddress.id_customer_address, {
+                                        label: 'page.checkout.address.delivery.address_select' | trans,
+                                        attr: {
+                                            class: embed.addressSelectClassName ~ ' ' ~ embed.deliveryContainerClassName ~ ' ' ~ embed.shipmentTypeHiddenClassName ~ ' ' ~ embed.jsName ~ '__trigger--' ~ embed.index,
```
[the remark this answers is not in the document]
Comment (danilmoro3) [reply src/Pyz/Yves/CheckoutPage/Theme/default/components/molecules/address-item-form-field-list/address-item-form-field-list.twig:85]: They are duplicated in core as well 
Removed duplicate in demoshop

File: src/Pyz/Yves/CheckoutPage/Theme/default/components/molecules/address-item-form-field-list/address-item-form-field-list.twig [hunk 4]
```diff
@@ -56,36 +69,35 @@
...
                     {% endblock %}
 
                     {% block productConfiguration %}
                         {% widget 'ProductConfigurationCartItemDisplayWidget'
                             args [data.productItem]
                             with {
                             data: {
-                                wrapperClassName: config.name ~ '__product-configuration-options ' ~ config.name ~ '__product-configuration-options--mobile',
+                                wrapperClassName: config.name ~ '__product-configuration-options',
                                 hasStatus: false,
```
[the remark this answers is not in the document]
Comment (danilmoro3) [reply src/Pyz/Yves/CheckoutPage/Theme/default/components/molecules/address-item-form-field-list/address-item-form-field-list.twig:99]: Yes, like product options or attributes

File: src/Pyz/Yves/ShopUi/Theme/default/components/organisms/side-drawer/side-drawer.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        if (this.shouldCloseByOverlayClick) {
+            this.mapOverlayClickEvent();
+        }
+    }
+
+    protected mapOverlayClickEvent(): void {
+        this.overlay.addEventListener('click', () => this.toggle(false));
+    }
+
+    toggle(isShownFromOutside?: boolean): void {
```
[the remark this answers is not in the document]
Comment (danilmoro3) [reply src/Pyz/Yves/ShopUi/Theme/default/components/organisms/side-drawer/side-drawer.ts:24]: Copy-past from core 
Made it protected

File: src/Pyz/Yves/CheckoutPage/Theme/default/views/address/address.twig [hunk 6]
```diff
@@ -39,126 +55,205 @@
...
-                        first_name: 'col col--sm-12 col--lg-4',
-                        last_name: 'col col--sm-12 col--lg-5',
-                        address1: 'col col--sm-12 col--md-8 col--lg-9',
-                        address2: 'col col--sm-12 col--md-4 col--lg-3',
-                        zip_code: 'col col--sm-12 col--lg-3',
-                        city: 'col col--sm-12 col--lg-5',
-                        iso2_code: 'col col--sm-12 col--lg-4',
+            <div class="{{ deliveryContainerClassName }} {{ shipmentTypeHiddenClassName }} col col--sm-12">
+                {% include molecule('form') with {
+                    class: shippingClassName ~ ' ' ~ shippingValidationContainerClassName ~ ' ' ~ (isAddressSelected ? hiddenClassName),
```
[the remark this answers is not in the document]
Comment (danilmoro3) [reply src/Pyz/Yves/CheckoutPage/Theme/default/views/address/address.twig:111]: Moved variables out of condition

File: src/Pyz/Yves/ShopUi/Theme/default/components/molecules/main-popup/main-popup.scss [hunk 7]
```diff
@@ -0,0 +1,66 @@
+@include shop-ui-main-popup {
+    $name: '.main-popup';
+    border-radius: rem(3);
+    padding: rem(20);
```
[the remark this answers is not in the document]
Comment (danilmoro3) [reply src/Pyz/Yves/ShopUi/Theme/default/components/molecules/main-popup/main-popup.scss:4]: height and width works as expected

File: src/Pyz/Yves/CheckoutPage/Theme/default/views/address/address.twig [hunk 8]
```diff
@@ -24,10 +25,36 @@
...
+            {% set multishipmentValidatorClassName = embed.jsAddressClass ~ '__multishipment-validate-next-checkout-step' %}
+            {% set billingSameAsShippingContainerClassName = embed.jsAddressClass ~ '__billing-same-as-shipping-container' %}
+            {% set billingSameAsShippingClassName = embed.jsAddressClass ~ '__billing-same-as-shipping' %}
+            {% set selectBillingAddressClassName = embed.jsAddressClass ~ '__form-select-billing-address' %}
+            {% set billingClassName = embed.jsAddressClass ~ '__billing' %}
+            {% set billingFormClassName = embed.jsAddressClass ~ '__billing-form' %}
+            {% set hiddenClassName = 'is-hidden' %}
+
+            {% set shippingAddress = data.form.shippingAddress.id_customer_address %}
+            {% set shippingAddressValue = shippingAddress.vars.value %}
```
Comment (FRITS-Kh) [inline src/Pyz/Yves/CheckoutPage/Theme/default/views/address/address.twig:49]: The `shippingAddress` can be undefined and it leads to an error in the `shippingAddressValue`.
Please check the workflow with a customer without saved addresses.

File: src/Pyz/Yves/CheckoutPage/Theme/default/views/address/address.twig [hunk 9]
```diff
@@ -38,127 +65,196 @@
...
+                embed: {
+                    jsAddressClass: embed.jsAddressClass,
+                    hiddenClassName: hiddenClassName,
+                    billingFormClassName: billingFormClassName,
+                    selectBillingAddressClassName: selectBillingAddressClassName,
+                },
+            } only %}
+                {% block fields %}
+                    {% set hasBillingAddresses = data.form.id_customer_address is defined %}
+                    {% set billingAddressValue = data.form.id_customer_address.vars.value %}
```
Comment (FRITS-Kh) [inline src/Pyz/Yves/CheckoutPage/Theme/default/views/address/address.twig:190]: The same, can't get a value from undefined.

**Review by FRITS-Kh — CHANGES_REQUESTED** [review]

**Review by FRITS-Kh — APPROVED** [review]

**Review by abitskil — APPROVED** [review]

**Pull request merged**